core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -230,14 +230,21 @@ private boolean isWriteSetOverlappedWith(Scan scan) {
     }
 
     for (Map.Entry<Key, Put> entry : writeSet.entrySet()) {
-      Put put = entry.getValue();
+      if (scanSet.get(scan).contains(entry.getKey())) {
+        return true;
+      }
 
+      Put put = entry.getValue();
       if (!put.forNamespace().equals(scan.forNamespace())
           || !put.forTable().equals(scan.forTable())
           || !put.getPartitionKey().equals(scan.getPartitionKey())) {
         continue;
       }
 
+      if (!areConjunctionsOverlapped(put, scan)) {
```

Comment:
I tried to remove this condition and ran the integration test. But any test case didn't fail. So, maybe there is room to add a test case for this?

Reply:
Definitely. We can test the case where clustering keys are overlapped, but conjunctions are not overlapped with put (it should not throw exceptions). Good catch, thank you!

Reply:
Added in 383f30d.